File: common/src/main/java/dev/obscuria/fragmentum/core/v1/common/event/BaseEventToken.java
package dev.obscuria.fragmentum.core.v1.common.event;

import dev.obscuria.fragmentum.api.v1.common.event.Event;
import dev.obscuria.fragmentum.api.v1.common.event.EventToken;
import org.jetbrains.annotations.ApiStatus;

import java.util.UUID;

@ApiStatus.Internal
public record BaseEventToken(UUID uuid, Event<?> event) implements EventToken
{
    @Override
    public void unregister()
    {
        event.unregister(this);
    }
}

File: common/src/main/kotlin/dev/obscuria/fragmentum/core/tools/event/BaseEventToken.kt
package dev.obscuria.fragmentum.core.tools.event

import dev.obscuria.fragmentum.api.tools.event.Event
import dev.obscuria.fragmentum.api.tools.event.EventToken
import java.util.*

@JvmRecord
data class BaseEventToken(val uuid: UUID, val event: Event<*>) : EventToken
{
    override fun unregister()
    {
        event.unregister(this)
    }
}
